### PR TITLE
support `ceiling` function

### PR DESCRIPTION
**Pull request by dseisun-materialize**
Adds support for ceiling function as an alias to the implementation of the ceil function

This solves: https://github.com/MaterializeInc/materialize/issues/15793

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way) and therefore is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [x] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  - Adds support for the `ceiling` function
  - Adds reference to the `ceiling` function in docs

### CODE REVIEW

File: test/sqllogictest/funcs.slt [hunk 1]
```diff
@@ -288,6 +288,11 @@ SELECT ceil(1.1), ceil(1.111), ceil(100.1), ceil(100.11), ceil(-4.1)
 ----
 2  2  101  101  -4
 
+query RRRRR
+SELECT ceiling(1.1), ceiling(1.111), ceiling(100.1), ceiling(100.11), ceiling(-4.1)
+----
+2  2  101  101  -3
```
Comment (mjibson) [inline test/sqllogictest/funcs.slt:294]: The -3 looks wrong here. Should be -4?

**Review by mjibson — APPROVED** [review]

File: test/sqllogictest/funcs.slt (hunk 1 above)
Comment (dseisun-materialize) [reply test/sqllogictest/funcs.slt:294]: wow, yeah. Fixing

File: test/sqllogictest/funcs.slt [hunk 2]
```diff
@@ -288,6 +288,11 @@ SELECT ceil(1.1), ceil(1.111), ceil(100.1), ceil(100.11), ceil(-4.1)
 ----
 2  2  101  101  -4
 
+query RRRRR
```
Comment (aljoscha) [inline test/sqllogictest/funcs.slt:291]: There are also existing (and commented out) tests for `ceiling` in `float8.slt` already: https://github.com/MaterializeInc/materialize/blob/b20c36cb926d6d88c0f797557eb92c496fa4bbd7/test/sqllogictest/postgres/float8.slt#L235

We should now uncomment these and remove the TODO.

**Review by joacoc — APPROVED** [review]

File: test/sqllogictest/funcs.slt (hunk 2 above)
Comment (dseisun-materialize) [reply test/sqllogictest/funcs.slt:291]: good catch. For my edification did you just grep for the term `ceiling` to find this?

Comment (aljoscha) [reply test/sqllogictest/funcs.slt:291]: I did! :sweat_smile:

**Comment by aljoscha** [comment]
Seems this PR is borked?

**Pull request closed without merging**